content/en-us/reference/engine/classes/Plugin.yaml

```diff
@@ -4,9 +4,58 @@ category: Plugins
 memory_category: Instances
 summary: ''
 description: |
-  Plugin is the main object responsible for creating basic studio widgets. It is
-  a custom add-on to Studio which adds new behavior and features that are not
-  normally included.
+  Plugin is the main object responsible for creating basic studio widgets,
+  plugin toolbars, plugin buttons and more. It is a custom add-on to Studio which
+  adds new behavior and features that are not normally included.
+
+  The Plugin object can be accessed through the `Global.RobloxGlobals.plugin` global
+  reference in a `Class.Script` that is executed as a plugin.
+
+  <Alert severity="warning">
+  The `Global.RobloxGlobals.plugin` global reference is not passed to `Class.ModuleScript`s within the plugin. In order to use it in a `Class.ModuleScript`, you must explicitly pass it as seen in the example below.
+  </Alert>
+
+  ```lua title='Script - Passing the plugin global to a module script.'
+  assert(plugin, "This script must be run as a plugin.")
+  -- Code beyond this point will execute only if the script is run as a plugin.
+  
+  -- Load the module and pass the plugin reference.
+  local pluginModule = require(script.Parent.PluginModule)
+  pluginModule:Initialize(plugin)
+  
+  -- Verify if the plugin reference was initialized.
+  pluginModule:CheckForPluginGlobal()
+  ```
+
+  ```lua title='ModuleScript - Recieving and storing the plugin global.'
+  local pluginModule = {}
+  local plugin -- Local plugin reference
+  
+  -- Initialize the plugin reference if not already set.
+  function pluginModule:Initialize(pluginReference: Plugin)
+      if plugin ~= pluginReference then
+          plugin = pluginReference
+      else
+          error("Plugin is already initialized.")
+      end
+  end
+  
+  -- Check if the plugin reference is set and print out appropriate info.
+  function pluginModule:CheckForPluginGlobal()
+      if plugin ~= nil then
+          print("Plugin reference is set!")
+      else
+          warn("Plugin reference is missing!")
+      end
+  end
+  
+  return pluginModule
+  ```
+
+  ```lua title='Output'
+    -- Plugin reference is set!
+  ```
+
 code_samples:
```

Comment:
```suggestion
  **Plugin** is the main object responsible for creating basic Studio widgets,
  plugin toolbars, plugin buttons, and more. It is a custom add-on to Studio
  which adds new behavior and features that are not normally included. The
  **Plugin** object can be accessed through the `Global.RobloxGlobals.plugin`
  global reference in a `Class.Script` that is executed as a plugin.
code_samples:
  - Plugin-Global-Pass
  - Plugin-Global-Receive
```

Reply:
In a separate PR, I made your code samples into linkable files and put them in the `code_samples` section.